### PR TITLE
Update wordpress/editor store calls to import approach

### PR DESCRIPTION
**Pull request by noahtallen**
## Proposed Changes
**Note: typecheck is currently failing, but the relevant package will be removed soon. (#75269)**

Related to #74473. The benefits to importing stores directly are:

1. Explicit dependencies, not implicit dependencies.
2. Better types (where applicable).

So this makes sure we reference the 'core/editor` store via an import instead of a string.

The main motivation for this PR was to fix some deprecation warnings on trunk. (Some selectors were moved to core/block-editor instead, so those were also updated.)

## Testing Instructions
Essentially, smoke test as many ETK features as you can, after sandboxing a test site and syncing ETK to your sandbox. This is pretty straightforward, so you can also verify I didn't make any typos :P.

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx [hunk 1]
```diff
@@ -53,8 +54,8 @@ export function PagePatternsPlugin( props: PagePatternsPluginProps ) {
 
 	const { getMeta, postContentBlock } = useSelect( ( select ) => {
 		const getMeta = () =>
-			( select( 'core/editor' ) as CoreEditorPlaceholder ).getEditedPostAttribute( 'meta' );
-		const currentBlocks = ( select( 'core/editor' ) as CoreEditorPlaceholder ).getBlocks();
+			( select( editorStore ) as CoreEditorPlaceholder ).getEditedPostAttribute( 'meta' );
+		const currentBlocks = ( select( 'core/block-editor' ) as CoreEditorPlaceholder ).getBlocks();
```
Comment (noahtallen) [inline apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx:58]: Basically, selectors and actions related to blocks were moved to core/block-editor. In the future, we should also do this for core/block-editor

File: .yarnrc.yml [hunk 2]
```diff
@@ -37,6 +37,10 @@ logFilters:
     pattern: "@automattic/wpcom-editing-toolkit@workspace:apps/editing-toolkit provides react (*) with version 17.0.2, which doesn't satisfy what @wordpress/block-editor and some of its descendants request"
   - level: discard
     pattern: "@automattic/wpcom-editing-toolkit@workspace:apps/editing-toolkit provides react-dom (*) with version 17.0.2, which doesn't satisfy what @wordpress/block-editor and some of its descendants request"
+  - level: discard
```
Comment (noahtallen) [inline .yarnrc.yml:40]: can't wait until we don't get these any more

File: apps/editing-toolkit/package.json [hunk 3]
```diff
@@ -80,6 +80,7 @@
 		"@wordpress/dependency-extraction-webpack-plugin": "^4.5.0",
 		"@wordpress/dom-ready": "^3.22.0",
 		"@wordpress/edit-post": "^6.19.0",
+		"@wordpress/editor": "^13.7.0",
```
Comment (noahtallen) [inline apps/editing-toolkit/package.json:84]: Explicit dependencies ftw!

File: yarn.lock [hunk 4]
```diff
@@ -8459,6 +8714,65 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@wordpress/components@npm:^23.7.0":
```
Comment (noahtallen) [inline yarn.lock]: uh oh

**Comment by tyxla** [comment]
> Note: typecheck is currently failing, but the relevant package will be removed soon. (https://github.com/Automattic/wp-calypso/pull/75269)

Rebased after landing #75269.

File: apps/editing-toolkit/package.json [hunk 5]
```diff
@@ -126,6 +127,7 @@
 		"@testing-library/jest-dom": "^5.16.2",
 		"@testing-library/react": "^12.1.3",
 		"@types/node": "^18.11.18",
+		"@types/wordpress__editor": "^13.6.0",
```
Comment (tyxla) [inline apps/editing-toolkit/package.json:130]: Note that those types are still not loaded due to a forced noop package resolution here:

https://github.com/Automattic/wp-calypso/blob/8c09604f5f03362894e1ac1a947117192453d13e/package.json#L293

**Comment by tyxla** [comment]
I rebased and deduped, however, it seems like there are still failures. I won't have the chance to take a look at them today.

File: apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx (hunk 1 above)
Comment (tyxla) [reply apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx:58]: I think we should do that change in a separate PR; I'd say it's too easy to break something that way. I'd suggest keeping references to the previously used store, and then if we change to a different store, that would be done in a separate PR with less noise. WDYT?

Comment (noahtallen) [reply apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx:58]: That's reasonable 👍

**Pull request closed without merging**